src/pss/signing_key.rs

```diff
@@ -227,11 +232,68 @@ where
     type Error = pkcs8::Error;
 
     fn try_from(private_key_info: pkcs8::PrivateKeyInfo<'_>) -> pkcs8::Result<Self> {
-        private_key_info
-            .algorithm
-            .assert_algorithm_oid(ID_RSASSA_PSS)?;
+        verify_algorithm_id(&private_key_info.algorithm)?;
         RsaPrivateKey::try_from(private_key_info).map(Self::new)
     }
 }
 
 impl<D> ZeroizeOnDrop for SigningKey<D> where D: Digest {}
+
+impl<D> PartialEq for SigningKey<D>
+where
+    D: Digest,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.inner == other.inner && self.salt_len == other.salt_len
```

Comment:
Hmm, I just noticed the derived `Eq`/`PartialEq` on `RsaPrivateKey` are not constant-time, which seems bad, although not directly related to this PR except it's indirectly invoking it.